service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/ActivityServiceBean.java

```diff
@@ -398,14 +405,53 @@ public int getPreviousFishingActivity(int fishingActivityId) {
         log.info("Retrieve fishing activity from db: {}", fishingActivityId);
         FishingActivityEntity activityEntity = fishingActivityDao.getFishingActivityById(fishingActivityId, null);
         log.info("activityEntity received from db Id: {} typeCode: {} Date: {}", activityEntity.getId(), activityEntity.getTypeCode(), DateUtils.parseUTCDateToString(activityEntity.getCalculatedStartTime()));
-        return fishingActivityDao.getPreviousFishingActivityId(activityEntity);
+
+        FishingTripEntity fishingTrip = activityEntity.getFishingTrip();
+        Set<FishingActivityEntity> fishingActivities = fishingTrip.getFishingActivities();
+
+        List<FishingActivityEntity> fishingActivityEntityList = new ArrayList<>(fishingActivities);
+        fishingActivityEntityList.sort(Comparator.comparing(FishingActivityEntity::getOccurence).reversed());
```

Comment:
Couldn't you just skip reversing the list and do "return fishingActivityEntityList.get(index - 1).getId()" later, instead of  "+ 1"?. Pretty small optimization compared to the work you did here but hey :)